### DISCUSSION TITLE
Parameters in swagger docs

### DISCUSSION THREAD

**Discussion by StephenKThung** [BerriAI/litellm#4734] [Q&A] [answered]
Hi all!

My team is researching running a litellm server and is working on some internal documentation for it. We've noticed that when fastapi auto-generates the openapi spec for litellm-added endpoints (via inclusion of the router from litellm/proxy/proxy_server.py), only some of the parameters of the endpoints actually show up, e.g. the model parameter in /chat/completions:

![image](https://github.com/user-attachments/assets/34593e27-4fa5-4672-8935-e05f5bd0bde8)

Is there a full openapi spec available for the litellm openai proxy that we could directly inject instead? If so, are there tracked versions of the spec? From what we understand the spec should be effectively a superset of the openai openapi spec but we haven't found any documentation of a version it would be tied to. Thanks!

**Comment by ishaan-jaff** [comment, marked as answer]
hi @StephenKThung thanks for looking at LiteLLM 

- Here's the supported input params: https://docs.litellm.ai/docs/completion/input 
- Output: https://docs.litellm.ai/docs/completion/output 

>  If so, are there tracked versions of the spec

Not as yet but i'm happy to work with your team on this 

@StephenKThung could we hop on a call to understand what your team needs ? Link to my cal for your convenience here: https://calendly.com/d/4mp-gd3-k5k/berriai-1-1-onboarding-litellm-hosted-version?month=2024-07 

If it's easier my linkedin is here https://www.linkedin.com/in/reffajnaahsi/

**Reply by StephenKThung** [reply]
Thanks for the super fast reply @ishaan-jaff !

I think this is the information we need for right now but I'll be sure to reach out if there's more we need.
